tool_path_planner/src/utilities.cpp

```
@@ -484,6 +503,165 @@ ToolPaths addExtraPaths(const ToolPaths& tool_paths, double offset_distance)
   return new_tool_paths;
 }
 
+ToolPath splitByAxes(const ToolPathSegment& tool_path_segment)
+{
+  using Point = pcl::PointXYZ;
+  // Sanity check - tool path segment must not be empty
+  if (tool_path_segment.size() == 0)
+  {
+    ROS_ERROR_STREAM("Tool path segment 0 is empty.");
+    // TODO: throw an exception
```

This does not seem worthy of an exception (which you didn't add anyway).  If a segment is of zero length, just immediately return an empty ToolPath.

---

I personally disagree. An empty toolpath should not be provided to an edge path splitter. I think this should throw an exception. Returning an empty toolpath requires the calling code to check for an empty toolpath, which it has already **not** done by passing an empty toolpath into this function.